docker/ci/build

```diff
@@ -15,23 +16,37 @@ function build_and_test_image() {
   : ===
   : === build and test $container_name
   : ===
-  bundle exec puppet-docker build "$container_name" --repository pcr-internal.puppet.net/release-engineering
+  bundle exec puppet-docker build "$container_name" --no-cache --repository pcr-internal.puppet.net/release-engineering --version "$version" --build-arg postgres_version="$postgres_version" --no-latest
   bundle exec puppet-docker spec "$container_name"
 }
 
 function push_image() {
   container_name="$1"
+  container_version="$2"
   : ===
   : === push $container_name
   : ===
-  bundle exec puppet-docker push "$container_name" --repository pcr-internal.puppet.net/release-engineering
+  bundle exec puppet-docker push "$container_name" --repository pcr-internal.puppet.net/release-engineering --version "$container_version" --no-latest
 }
 
 : ===
 : === bundle install to get ready
 : ===
 bundle install --path "$BUNDLER_PATH"
 
+: ===
+: === If we do not git pull --unshallow we get really weird results with git describe
+: ===
+git pull --unshallow
+
+: ===
+: === make sure we fetch tags for versioning
+: ===
+git fetch origin 'refs/tags/*:refs/tags/*'
+git_describe=`git describe`
```

Comment:
I don't think that's going to be relevant here, it looks like `git describe --always` will either match the output of `git describe`, or it will give the git sha if it doesn't find a tag. The build will fail if the version is set to a git sha (since that's used to pick the version of puppetdb installed from repos), I think failing at the `git describe` would make it more straightforward to debug than failing in the container build